core/Scheduler/Task.php

```diff
@@ -72,22 +83,31 @@ class Task
      *                                          should be executed and how long before the next execution.
      * @param int $priority The priority of the task. Tasks with a higher priority will be executed first.
      *                      Tasks with low priority will be executed last.
+     * @param int $ttlInSeconds TTL to use for this task. Defauts to 3600. See {@link self::$ttlInSeconds}
      * @throws Exception
      */
     public function __construct($objectInstance, $methodName, $methodParameter, $scheduledTime,
-                                $priority = self::NORMAL_PRIORITY)
+                                $priority = self::NORMAL_PRIORITY, int $ttlInSeconds = null)
     {
         $this->className = $this->getClassNameFromInstance($objectInstance);
 
         if ($priority < self::HIGHEST_PRIORITY || $priority > self::LOWEST_PRIORITY) {
             throw new Exception("Invalid priority for ScheduledTask '$this->className.$methodName': $priority");
         }
 
+        $ttlInSeconds = $ttlInSeconds ?? 3600;
+
+        // only allow TTLs between 1 second and 1 week
+        if ($ttlInSeconds !== -1 && ($ttlInSeconds < 1 || $ttlInSeconds > 604800)) {
+            throw new Exception("Invalid ttl for ScheduledTask '$this->className.$methodName': $ttlInSeconds. The TTL must be -1 or between 1 and 604800.");
```

Comment:
Wherever the acronym stands on its own, it should be capitalised.
```suggestion
            throw new Exception("Invalid TTL for ScheduledTask '$this->className.$methodName': $ttlInSeconds. The TTL must be -1 or between 1 and 604800 (1 second to 1 week).");
```